src/modules/settings.ts

```diff
@@ -35,9 +42,9 @@ export function initCallbacks(): void {
     if(!isValidKey(key)){
       continue;
     }
-    
+
     uiElem.oninput = () => {
-      loadSettings()[key] = parseSetting(key, value) ?? defaultSettings[key];
```

Comment:
This I could not get to work with the new setting. Either the abstraction in this module is borked or I'm not experienced enough with TypeScript to get it work - I suspect the latter is the case. But at least this works and you get an idea what this feature looks like. Feel free to fix this.